app/modules/provisioning/groups.py

```diff
@@ -81,9 +81,18 @@ def log_groups(
         groups (list): The list of groups to log.
         group_display_key (str, optional): The key to display in the logs. Defaults to None.
     """
+    if not group_display_key:
+        logger.warn(f"{integration_name}No group display key provided.")
```

Comment:
Super nit picky but not sure if you want to put a space in there between the {integration_name} and No group.... so that the logs are more readable.